### ISSUE TITLE
gha-utilities/sass-build@v0.1.6 undefined:3 throw error;

### ISSUE THREAD

**Issue by christopher-paul-shaw** [gha-utilities/sass-build#11]
Unable to get any sass built. Even when example added to main.yml, constantly receiving same error

When running github action, when sass build starts running, persistently returns the same error

 Sass Build1s
##[error]Node run failed with exit code 1
Run gha-utilities/sass-build@v0.1.6
undefined:3
throw error;

1.  Add sass-build example to main,yml

2. Please format multi-line output with triple back-ticks, eg...

```bash
 Sass Build1s
##[error]Node run failed with exit code 1
Run gha-utilities/sass-build@v0.1.6
undefined:3
throw error;
^

Error: FormatException: NaN
    at Object._newRenderError (/home/runner/work/_actions/gha-utilities/sass-build/v0.1.6/node_modules/sass/sass.dart.js:10899:19)
    at StaticClosure._renderSync (/home/runner/work/_actions/gha-utilities/sass-build/v0.1.6/node_modules/sass/sass.dart.js:10724:18)
    at Object.Primitives_applyFunction (/home/runner/work/_actions/gha-utilities/sass-build/v0.1.6/node_modules/sass/sass.dart.js:1064:30)
    at Object.Function_apply (/home/runner/work/_actions/gha-utilities/sass-build/v0.1.6/node_modules/sass/sass.dart.js:4902:16)
    at _callDartFunctionFast (/home/runner/work/_actions/gha-utilities/sass-build/v0.1.6/node_modules/sass/sass.dart.js:6584:16)
    at Object.renderSync (/home/runner/work/_actions/gha-utilities/sass-build/v0.1.6/node_modules/sass/sass.dart.js:6562:18)
    at Object.<anonymous> (/home/runner/work/_actions/gha-utilities/sass-build/v0.1.6/index.js:111:21)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
    at Module.load (internal/modules/cjs/loader.js:815:32) {
  formatted: 'Error: FormatException: NaN',
  status: 3
}
##[error]Node run failed with exit code 1
```

**Comment by S0AndS0** [comment]
Thanks for opening this Issue!


It seems there where changes in how forgiving Sass is with undefined settings.

___


## TLDR


For now it's possible to get past this error by defining an `indentWidth`, eg...


```YAML
    - name: Compile main.css from main.scss
      uses: gha-utilities/sass-build@v0.1.6
      with:
        debug: true
        indentWidth: 2
        source: _scss/main.scss
        destination: /tmp/repo-name/assets/css/main.css
```


... I'll be adding defaults when this setting is undefined upon releasing `v0.1.7`


___


I can confirm/reproduce these errors with the following...


## Test setup


```Bash
cd ~/git/hub/gha-utilities/sass-build

mkdir -vp .tests/{lib,out}
```


## Test files


**`.tests/lib/something.scss`**

```SCSS
@mixin onething {
  @debug 'mixing something has been called';
}

.another-thing {
  width: auto;
}
```


**`.tests/main.scss`**


```SCSS
@use 'lib/something';

.example-class {
  width: auto;
  height: auto;
  @include something.onething;
}
```


## Test command


```Bash
INPUT_SOURCE=.tests/main.scss\
 INPUT_DESTINATION=.tests/out/main.css\
 npm run test
```


## Error output


```
> sass-build@0.1.6 test ~/git/hub/gha-utilities/sass-build
> node index.js

undefined:1
throw error;
^

Error: FormatException: NaN
    at Object._newRenderError (~/git/hub/gha-utilities/sass-build/node_modules/sass/sass.dart.js:10899:19)
    at StaticClosure._renderSync (~/git/hub/gha-utilities/sass-build/node_modules/sass/sass.dart.js:10724:18)
    at Object.Primitives_applyFunction (~/git/hub/gha-utilities/sass-build/node_modules/sass/sass.dart.js:1064:30)
    at Object.Function_apply (~/git/hub/gha-utilities/sass-build/node_modules/sass/sass.dart.js:4902:16)
    at _callDartFunctionFast (~/git/hub/gha-utilities/sass-build/node_modules/sass/sass.dart.js:6584:16)
    at Object.renderSync (~/git/hub/gha-utilities/sass-build/node_modules/sass/sass.dart.js:6562:18)
    at Object.<anonymous> (~/git/hub/gha-utilities/sass-build/index.js:111:21)
    at Module._compile (internal/modules/cjs/loader.js:1151:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1171:10)
    at Module.load (internal/modules/cjs/loader.js:1000:32) {
  formatted: 'Error: FormatException: NaN',
  status: 3
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sass-build@0.1.6 test: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sass-build@0.1.6 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2020-02-14T23_05_31_332Z-debug.log
```

___


## Hints


It's possible to have this project output all `render_options` that are passed to `sass.renderSync` method by adding `debug: true` to the YAML file or `INPUT_DEBUG=true` to the `npm run test` command.


Example `main.yml` snip...


```YAML
    - name: Compile main.css from main.scss
      uses: gha-utilities/sass-build@v0.1.6
      with:
        debug: true
        source: _scss/main.scss
        destination: /tmp/repo-name/assets/css/main.css
```


Example `npm run test` command...


```Bash
INPUT_DEBUG=true\
 INPUT_SOURCE=.tests/main.scss\
 INPUT_DESTINATION=.tests/out/main.css\
 npm run test
```

___


## Purposed fix tests


```Bash
INPUT_INDENTWIDTH=2\
 INPUT_OUTPUTSTYLE=expanded\
 INPUT_SOURCE=.tests/main.scss\
 INPUT_DESTINATION=.tests/out/main.css\
 npm run test
```

Example output...


```
> sass-build@0.1.6 test ~/git/hub/gha-utilities/sass-build
> node index.js

--- render_options ---
┌─────────────┬────────────────────┐
│   (index)   │       Values       │
├─────────────┼────────────────────┤
│    file     │ '.tests/main.scss' │
│ outputStyle │     'expanded'     │
│  precision  │        NaN         │
│ indentWidth │         2          │
└─────────────┴────────────────────┘
.tests/lib/something.scss:2 DEBUG: mixing something has been called
Wrote file -> .tests/out/main.css
```


**`.tests/out/main.css`**


```CSS
.another-thing {
  width: auto;
}

.example-class {
  width: auto;
  height: auto;
}
```

**Comment by S0AndS0** [comment]
I'll be closing this Issue in the next day or two unless I read-back from ya that `v0.1.7` hasn't resolved errors.

Thanks again for opening this Issue, and feel free to open new Issues for new bugs or re-open this one if some regression causes these errors to pop again.
